File: frontend/src/services/bankPayrollService.js
import apiClient from './api';

export const getBankBatches = () => apiClient.get('/suite/bank-payroll/batches');
export const createBatch = (dto) => apiClient.post('/suite/bank-payroll/batches', dto);
export const exportFile = (batchId, format) => apiClient.post(`/suite/bank-payroll/batches/${batchId}/export?format=${format}`);

File: backend/src/main/java/com/awais/hr/module/bankpayroll/service/BankPayrollService.java
package com.awais.hr.module.bankpayroll.service;

import java.util.List;
import java.util.Map;

public interface BankPayrollService {
    List<Map<String, Object>> getBankBatches();
    Map<String, Object> createBatch(Map<String, Object> body);
    Map<String, Object> exportFile(String batchId, String format);
}
